iris/src/iris/pipeline/delete_lecture_units_pipeline.py

```diff
@@ -0,0 +1,183 @@
+        self.transcription_collection = init_lecture_transcription_schema(client)
+        self.lecture_unit_segment_summary_collection = init_lecture_unit_segment_schema(
+            client
+        )
+        self.lecture_unit_collection = init_lecture_unit_schema(client)
+        self.lecture_units = lecture_units
+        self.artemis_base_url = artemis_base_url
+        self.callback = callback
+
+    def __call__(self) -> None:
+        self.callback.in_progress("deleting lecture units...")
+        self.delete_entries_for_lecture_units()
+        self.callback.done("lecture unit deletion done")
+
+    def delete_entries_for_lecture_units(self):
+        try:
+            for lecture_unit in self.lecture_units:
+                if self.delete_page_chunk(lecture_unit):
+                    logger.info("Page chunks deleted successfully")
+                else:
+                    logger.error("Failed to delete page chunks")
+
+                if self.delete_transcriptions(lecture_unit):
+                    logger.info("Transcriptions deleted successfully")
+                else:
+                    logger.info("Failed to delete transcriptions")
+
+                if self.delete_lecture_unit_segments(lecture_unit):
+                    logger.info("Lecture unit segments deleted successfully")
+                else:
+                    logger.info("Failed to delete Lecture unit segments")
+
+                if self.delete_lecture_unit(lecture_unit):
+                    logger.info("Lecture unit deleted successfully")
+                else:
+                    logger.info("Failed to delete lecture unit")
+            self.callback.done("Lecture unit removed")
+        except Exception as e:
+            logger.error("Error deleting lecture unit: %s", e)
+            self.callback.error("Error while removing old slides")
+            return False
+
+    def delete_page_chunk(self, lecture_unit: LectureUnitPageDTO):
+        """
+        Delete the page chunks from the database
+        """
+        try:
+            self.page_chunk_collection.data.delete_many(
+                where=Filter.by_property(
+                    LectureUnitPageChunkSchema.BASE_URL.value
+                ).equal(self.artemis_base_url)
+                & Filter.by_property(LectureUnitPageChunkSchema.COURSE_ID.value).equal(
+                    lecture_unit.course_id
+                )
+                & Filter.by_property(LectureUnitPageChunkSchema.LECTURE_ID.value).equal(
+                    lecture_unit.lecture_id
+                )
+                & Filter.by_property(
+                    LectureUnitPageChunkSchema.LECTURE_UNIT_ID.value
+                ).equal(lecture_unit.lecture_unit_id)
+            )
+            return True
+        except Exception as e:
+            logger.error(
+                "Error deleting lecture unit page chunks: %s", e, exc_info=True
+            )
+            return False
```

Comment:
Looking at all delete methods I would propose adding only one delete function and calling it for the page chunks for example like this:
```suggestion
    def _delete_with_filter(self, collection, schema, lecture_unit: LectureUnitPageDTO, log_context: str):
    try:
        collection.data.delete_many(
            where=Filter.by_property(schema.BASE_URL.value).equal(self.artemis_base_url)
            & Filter.by_property(schema.COURSE_ID.value).equal(lecture_unit.course_id)
            & Filter.by_property(schema.LECTURE_ID.value).equal(lecture_unit.lecture_id)
            & Filter.by_property(schema.LECTURE_UNIT_ID.value).equal(lecture_unit.lecture_unit_id)
        )
        logger.info(f"{log_context} deleted successfully")
        return True
    except Exception as e:
        logger.error(f"Error deleting {log_context}: %s", e, exc_info=True)
        return False
        
         def delete_page_chunk(self, lecture_unit: LectureUnitPageDTO):
         
    # insert above:
    return self._delete_with_filter(
        self.page_chunk_collection,
        LectureUnitPageChunkSchema,
        lecture_unit,
        "Page chunks"
    )
```